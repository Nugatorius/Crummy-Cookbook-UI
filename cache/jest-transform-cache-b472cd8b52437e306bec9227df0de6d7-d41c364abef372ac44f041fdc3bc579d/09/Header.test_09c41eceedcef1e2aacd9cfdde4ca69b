faf25d2a4bffdeab039e291a90fe76d2
var _reactNative=require('react-native');var _react=require('react');var _react2=_interopRequireDefault(_react);var _=require('../');var _2=_interopRequireDefault(_);var _reactTestRenderer=require('react-test-renderer');var _reactTestRenderer2=_interopRequireDefault(_reactTestRenderer);var _enzyme=require('enzyme');var _enzyme2=_interopRequireDefault(_enzyme);var _enzymeAdapterReact=require('enzyme-adapter-react-16');var _enzymeAdapterReact2=_interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}_enzyme2.default.configure({adapter:new _enzymeAdapterReact2.default()});describe('Header',function(){it('renders correctly',function(){var tree=_reactTestRenderer2.default.create(_react2.default.createElement(_2.default,null)).toJSON();expect(tree).toMatchSnapshot();});it('it should render 1 text component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.Text,null));expect(wrapper).toHaveLength(1);});it('it should render 1 view component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.View,null));expect(wrapper).toHaveLength(1);});it('it should render title as hello when given prop',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_2.default,{title:'hello'}));expect(wrapper.find('Text').text()).toEqual('hello');});});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImNvbmZpZ3VyZSIsImFkYXB0ZXIiLCJkZXNjcmliZSIsIml0IiwidHJlZSIsImNyZWF0ZSIsInRvSlNPTiIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCIsIndyYXBwZXIiLCJ0b0hhdmVMZW5ndGgiLCJmaW5kIiwidGV4dCIsInRvRXF1YWwiXSwibWFwcGluZ3MiOiJBQUFBLHlDQUVBLDRCLDJDQUNBLHFCLGlDQUVBLHNELG1FQUNBLDhCLDZDQUVBLDJELHdKQUNBLGlCQUFPQSxTQUFQLENBQWlCLENBQUVDLFFBQVMsa0NBQVgsQ0FBakIsRUFFQUMsU0FBUyxRQUFULENBQW1CLFVBQU0sQ0FDdkJDLEdBQUcsbUJBQUgsQ0FBd0IsVUFBTSxDQUM1QixHQUFNQyxNQUFPLDRCQUFTQyxNQUFULENBQ1gsOENBRFcsRUFFWEMsTUFGVyxFQUFiLENBR0FDLE9BQU9ILElBQVAsRUFBYUksZUFBYixHQUNELENBTEQsRUFNQUwsR0FBRyxtQ0FBSCxDQUF3QyxVQUFNLENBQzVDLEdBQU1NLFNBQVUsb0JBQVEscURBQVIsQ0FBaEIsQ0FDQUYsT0FBT0UsT0FBUCxFQUFnQkMsWUFBaEIsQ0FBNkIsQ0FBN0IsRUFDRCxDQUhELEVBSUFQLEdBQUcsbUNBQUgsQ0FBd0MsVUFBTSxDQUM1QyxHQUFNTSxTQUFVLG9CQUFRLHFEQUFSLENBQWhCLENBQ0FGLE9BQU9FLE9BQVAsRUFBZ0JDLFlBQWhCLENBQTZCLENBQTdCLEVBQ0QsQ0FIRCxFQUlBUCxHQUFHLGlEQUFILENBQXNELFVBQU0sQ0FDMUQsR0FBTU0sU0FBVSxvQkFBUSwwQ0FBUSxNQUFNLE9BQWQsRUFBUixDQUFoQixDQUNBRixPQUFPRSxRQUFRRSxJQUFSLENBQWEsTUFBYixFQUFxQkMsSUFBckIsRUFBUCxFQUFvQ0MsT0FBcEMsQ0FBNEMsT0FBNUMsRUFDRCxDQUhELEVBSUQsQ0FuQkQiLCJmaWxlIjoiSGVhZGVyLnRlc3QuanMiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgJ3JlYWN0LW5hdGl2ZSc7XG5pbXBvcnQgeyBUZXh0LCBWaWV3IH0gZnJvbSAncmVhY3QtbmF0aXZlJztcbmltcG9ydCBSZWFjdCBmcm9tICdyZWFjdCc7XG5pbXBvcnQgSGVhZGVyIGZyb20gJy4uLyc7XG5cbmltcG9ydCByZW5kZXJlciBmcm9tICdyZWFjdC10ZXN0LXJlbmRlcmVyJztcbmltcG9ydCB7IHNoYWxsb3cgfSBmcm9tICdlbnp5bWUnO1xuaW1wb3J0IEVuenltZSBmcm9tICdlbnp5bWUnO1xuaW1wb3J0IEVuenltZUFkYXB0ZXIgZnJvbSAnZW56eW1lLWFkYXB0ZXItcmVhY3QtMTYnO1xuRW56eW1lLmNvbmZpZ3VyZSh7IGFkYXB0ZXI6IG5ldyBFbnp5bWVBZGFwdGVyKCkgfSk7XG5cbmRlc2NyaWJlKCdIZWFkZXInLCAoKSA9PiB7XG4gIGl0KCdyZW5kZXJzIGNvcnJlY3RseScsICgpID0+IHtcbiAgICBjb25zdCB0cmVlID0gcmVuZGVyZXIuY3JlYXRlKFxuICAgICAgPEhlYWRlci8+XG4gICAgKS50b0pTT04oKTtcbiAgICBleHBlY3QodHJlZSkudG9NYXRjaFNuYXBzaG90KCk7XG4gIH0pO1xuICBpdCgnaXQgc2hvdWxkIHJlbmRlciAxIHRleHQgY29tcG9uZW50JywgKCkgPT4ge1xuICAgIGNvbnN0IHdyYXBwZXIgPSBzaGFsbG93KDxUZXh0Lz4pO1xuICAgIGV4cGVjdCh3cmFwcGVyKS50b0hhdmVMZW5ndGgoMSk7XG4gIH0pO1xuICBpdCgnaXQgc2hvdWxkIHJlbmRlciAxIHZpZXcgY29tcG9uZW50JywgKCkgPT4ge1xuICAgIGNvbnN0IHdyYXBwZXIgPSBzaGFsbG93KDxWaWV3Lz4pO1xuICAgIGV4cGVjdCh3cmFwcGVyKS50b0hhdmVMZW5ndGgoMSk7XG4gIH0pO1xuICBpdCgnaXQgc2hvdWxkIHJlbmRlciB0aXRsZSBhcyBoZWxsbyB3aGVuIGdpdmVuIHByb3AnLCAoKSA9PiB7XG4gICAgY29uc3Qgd3JhcHBlciA9IHNoYWxsb3coPEhlYWRlciB0aXRsZT0naGVsbG8nLz4pO1xuICAgIGV4cGVjdCh3cmFwcGVyLmZpbmQoJ1RleHQnKS50ZXh0KCkpLnRvRXF1YWwoJ2hlbGxvJyk7XG4gIH0pO1xufSk7XG4iXX0=