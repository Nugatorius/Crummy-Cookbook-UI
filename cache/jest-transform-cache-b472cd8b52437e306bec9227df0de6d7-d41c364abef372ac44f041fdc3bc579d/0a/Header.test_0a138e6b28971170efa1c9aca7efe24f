db7d6ca88229a777e4bd557e524e6983
var _reactNative=require('react-native');var _react=require('react');var _react2=_interopRequireDefault(_react);var _=require('../');var _2=_interopRequireDefault(_);var _reactTestRenderer=require('react-test-renderer');var _reactTestRenderer2=_interopRequireDefault(_reactTestRenderer);var _enzyme=require('enzyme');var _enzyme2=_interopRequireDefault(_enzyme);var _enzymeAdapterReact=require('enzyme-adapter-react-16');var _enzymeAdapterReact2=_interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}_enzyme2.default.configure({adapter:new _enzymeAdapterReact2.default()});describe('Header',function(){it('renders correctly',function(){var tree=_reactTestRenderer2.default.create(_react2.default.createElement(_2.default,{title:'Header'})).toJSON();console.log(tree);});it('it should render 1 text component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.Text,null));expect(wrapper).toHaveLength(1);});it('it should render 1 view component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.View,null));expect(wrapper).toHaveLength(1);});});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImNvbmZpZ3VyZSIsImFkYXB0ZXIiLCJkZXNjcmliZSIsIml0IiwidHJlZSIsImNyZWF0ZSIsInRvSlNPTiIsImNvbnNvbGUiLCJsb2ciLCJ3cmFwcGVyIiwiZXhwZWN0IiwidG9IYXZlTGVuZ3RoIl0sIm1hcHBpbmdzIjoiQUFBQSx5Q0FFQSw0QiwyQ0FDQSxxQixpQ0FFQSxzRCxtRUFDQSw4Qiw2Q0FFQSwyRCx3SkFDQSxpQkFBT0EsU0FBUCxDQUFpQixDQUFFQyxRQUFTLGtDQUFYLENBQWpCLEVBRUFDLFNBQVMsUUFBVCxDQUFtQixVQUFNLENBQ3ZCQyxHQUFHLG1CQUFILENBQXdCLFVBQU0sQ0FDNUIsR0FBTUMsTUFBTyw0QkFBU0MsTUFBVCxDQUNYLDBDQUFRLE1BQU0sUUFBZCxFQURXLEVBRVhDLE1BRlcsRUFBYixDQUdBQyxRQUFRQyxHQUFSLENBQVlKLElBQVosRUFFRCxDQU5ELEVBT0FELEdBQUcsbUNBQUgsQ0FBd0MsVUFBTSxDQUM1QyxHQUFNTSxTQUFVLG9CQUFRLHFEQUFSLENBQWhCLENBQ0FDLE9BQU9ELE9BQVAsRUFBZ0JFLFlBQWhCLENBQTZCLENBQTdCLEVBQ0QsQ0FIRCxFQUlBUixHQUFHLG1DQUFILENBQXdDLFVBQU0sQ0FDNUMsR0FBTU0sU0FBVSxvQkFBUSxxREFBUixDQUFoQixDQUNBQyxPQUFPRCxPQUFQLEVBQWdCRSxZQUFoQixDQUE2QixDQUE3QixFQUNELENBSEQsRUFJRCxDQWhCRCIsImZpbGUiOiJIZWFkZXIudGVzdC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCB7IFRleHQsIFZpZXcgfSBmcm9tICdyZWFjdC1uYXRpdmUnO1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBIZWFkZXIgZnJvbSAnLi4vJztcblxuaW1wb3J0IHJlbmRlcmVyIGZyb20gJ3JlYWN0LXRlc3QtcmVuZGVyZXInO1xuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XG5pbXBvcnQgRW56eW1lIGZyb20gJ2VuenltZSc7XG5pbXBvcnQgRW56eW1lQWRhcHRlciBmcm9tICdlbnp5bWUtYWRhcHRlci1yZWFjdC0xNic7XG5Fbnp5bWUuY29uZmlndXJlKHsgYWRhcHRlcjogbmV3IEVuenltZUFkYXB0ZXIoKSB9KTtcblxuZGVzY3JpYmUoJ0hlYWRlcicsICgpID0+IHtcbiAgaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICAgIGNvbnN0IHRyZWUgPSByZW5kZXJlci5jcmVhdGUoXG4gICAgICA8SGVhZGVyIHRpdGxlPSdIZWFkZXInLz5cbiAgICApLnRvSlNPTigpO1xuICAgIGNvbnNvbGUubG9nKHRyZWUpO1xuICAgIC8vIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbiAgfSk7XG4gIGl0KCdpdCBzaG91bGQgcmVuZGVyIDEgdGV4dCBjb21wb25lbnQnLCAoKSA9PiB7XG4gICAgY29uc3Qgd3JhcHBlciA9IHNoYWxsb3coPFRleHQvPik7XG4gICAgZXhwZWN0KHdyYXBwZXIpLnRvSGF2ZUxlbmd0aCgxKTtcbiAgfSk7XG4gIGl0KCdpdCBzaG91bGQgcmVuZGVyIDEgdmlldyBjb21wb25lbnQnLCAoKSA9PiB7XG4gICAgY29uc3Qgd3JhcHBlciA9IHNoYWxsb3coPFZpZXcvPik7XG4gICAgZXhwZWN0KHdyYXBwZXIpLnRvSGF2ZUxlbmd0aCgxKTtcbiAgfSk7XG59KTtcbiJdfQ==