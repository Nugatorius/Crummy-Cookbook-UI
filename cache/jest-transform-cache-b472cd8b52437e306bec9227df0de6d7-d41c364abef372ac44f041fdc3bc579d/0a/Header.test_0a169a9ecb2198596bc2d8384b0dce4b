6533d0ceb8067b0d959364990e33e8f0
var _reactNative=require('react-native');var _react=require('react');var _react2=_interopRequireDefault(_react);var _=require('../');var _2=_interopRequireDefault(_);var _reactTestRenderer=require('react-test-renderer');var _reactTestRenderer2=_interopRequireDefault(_reactTestRenderer);var _enzyme=require('enzyme');var _enzyme2=_interopRequireDefault(_enzyme);var _enzymeAdapterReact=require('enzyme-adapter-react-16');var _enzymeAdapterReact2=_interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}_enzyme2.default.configure({adapter:new _enzymeAdapterReact2.default()});describe('Header',function(){it('renders correctly',function(){var tree=_reactTestRenderer2.default.create(_react2.default.createElement(_2.default,{title:'Header'})).toJSON();console.log(tree.children.children);});it('it should render 1 text component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.Text,null));expect(wrapper).toHaveLength(1);});it('it should render 1 view component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.View,null));expect(wrapper).toHaveLength(1);});});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImNvbmZpZ3VyZSIsImFkYXB0ZXIiLCJkZXNjcmliZSIsIml0IiwidHJlZSIsImNyZWF0ZSIsInRvSlNPTiIsImNvbnNvbGUiLCJsb2ciLCJjaGlsZHJlbiIsIndyYXBwZXIiLCJleHBlY3QiLCJ0b0hhdmVMZW5ndGgiXSwibWFwcGluZ3MiOiJBQUFBLHlDQUVBLDRCLDJDQUNBLHFCLGlDQUVBLHNELG1FQUNBLDhCLDZDQUVBLDJELHdKQUNBLGlCQUFPQSxTQUFQLENBQWlCLENBQUVDLFFBQVMsa0NBQVgsQ0FBakIsRUFFQUMsU0FBUyxRQUFULENBQW1CLFVBQU0sQ0FDdkJDLEdBQUcsbUJBQUgsQ0FBd0IsVUFBTSxDQUM1QixHQUFNQyxNQUFPLDRCQUFTQyxNQUFULENBQ1gsMENBQVEsTUFBTSxRQUFkLEVBRFcsRUFFWEMsTUFGVyxFQUFiLENBR0FDLFFBQVFDLEdBQVIsQ0FBWUosS0FBS0ssUUFBTCxDQUFjQSxRQUExQixFQUVELENBTkQsRUFPQU4sR0FBRyxtQ0FBSCxDQUF3QyxVQUFNLENBQzVDLEdBQU1PLFNBQVUsb0JBQVEscURBQVIsQ0FBaEIsQ0FDQUMsT0FBT0QsT0FBUCxFQUFnQkUsWUFBaEIsQ0FBNkIsQ0FBN0IsRUFDRCxDQUhELEVBSUFULEdBQUcsbUNBQUgsQ0FBd0MsVUFBTSxDQUM1QyxHQUFNTyxTQUFVLG9CQUFRLHFEQUFSLENBQWhCLENBQ0FDLE9BQU9ELE9BQVAsRUFBZ0JFLFlBQWhCLENBQTZCLENBQTdCLEVBQ0QsQ0FIRCxFQUlELENBaEJEIiwiZmlsZSI6IkhlYWRlci50ZXN0LmpzIiwic291cmNlc0NvbnRlbnQiOlsiaW1wb3J0ICdyZWFjdC1uYXRpdmUnO1xuaW1wb3J0IHsgVGV4dCwgVmlldyB9IGZyb20gJ3JlYWN0LW5hdGl2ZSc7XG5pbXBvcnQgUmVhY3QgZnJvbSAncmVhY3QnO1xuaW1wb3J0IEhlYWRlciBmcm9tICcuLi8nO1xuXG5pbXBvcnQgcmVuZGVyZXIgZnJvbSAncmVhY3QtdGVzdC1yZW5kZXJlcic7XG5pbXBvcnQgeyBzaGFsbG93IH0gZnJvbSAnZW56eW1lJztcbmltcG9ydCBFbnp5bWUgZnJvbSAnZW56eW1lJztcbmltcG9ydCBFbnp5bWVBZGFwdGVyIGZyb20gJ2VuenltZS1hZGFwdGVyLXJlYWN0LTE2JztcbkVuenltZS5jb25maWd1cmUoeyBhZGFwdGVyOiBuZXcgRW56eW1lQWRhcHRlcigpIH0pO1xuXG5kZXNjcmliZSgnSGVhZGVyJywgKCkgPT4ge1xuICBpdCgncmVuZGVycyBjb3JyZWN0bHknLCAoKSA9PiB7XG4gICAgY29uc3QgdHJlZSA9IHJlbmRlcmVyLmNyZWF0ZShcbiAgICAgIDxIZWFkZXIgdGl0bGU9J0hlYWRlcicvPlxuICAgICkudG9KU09OKCk7XG4gICAgY29uc29sZS5sb2codHJlZS5jaGlsZHJlbi5jaGlsZHJlbik7XG4gICAgLy8gZXhwZWN0KHRyZWUpLnRvTWF0Y2hTbmFwc2hvdCgpO1xuICB9KTtcbiAgaXQoJ2l0IHNob3VsZCByZW5kZXIgMSB0ZXh0IGNvbXBvbmVudCcsICgpID0+IHtcbiAgICBjb25zdCB3cmFwcGVyID0gc2hhbGxvdyg8VGV4dC8+KTtcbiAgICBleHBlY3Qod3JhcHBlcikudG9IYXZlTGVuZ3RoKDEpO1xuICB9KTtcbiAgaXQoJ2l0IHNob3VsZCByZW5kZXIgMSB2aWV3IGNvbXBvbmVudCcsICgpID0+IHtcbiAgICBjb25zdCB3cmFwcGVyID0gc2hhbGxvdyg8Vmlldy8+KTtcbiAgICBleHBlY3Qod3JhcHBlcikudG9IYXZlTGVuZ3RoKDEpO1xuICB9KTtcbn0pO1xuIl19