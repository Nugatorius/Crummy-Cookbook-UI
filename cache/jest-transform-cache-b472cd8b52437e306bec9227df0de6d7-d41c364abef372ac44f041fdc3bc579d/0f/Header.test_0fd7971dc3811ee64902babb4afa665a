322de32dbcb9ced9c8dc707fbc9a165c
var _reactNative=require('react-native');var _react=require('react');var _react2=_interopRequireDefault(_react);var _=require('../');var _2=_interopRequireDefault(_);var _reactTestRenderer=require('react-test-renderer');var _reactTestRenderer2=_interopRequireDefault(_reactTestRenderer);var _enzyme=require('enzyme');var _enzyme2=_interopRequireDefault(_enzyme);var _enzymeAdapterReact=require('enzyme-adapter-react-16');var _enzymeAdapterReact2=_interopRequireDefault(_enzymeAdapterReact);function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}_enzyme2.default.configure({adapter:new _enzymeAdapterReact2.default()});describe('Header',function(){it('renders correctly',function(){var tree=_reactTestRenderer2.default.create(_react2.default.createElement(_2.default,null)).toJSON();expect(tree).toMatchSnapshot();});it('it should render 1 text component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.Text,null));expect(wrapper).toHaveLength(1);});it('it should render 1 view component',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_reactNative.View,null));expect(wrapper).toHaveLength(1);});it('it should render title as hello when given prop',function(){var wrapper=(0,_enzyme.shallow)(_react2.default.createElement(_2.default,{title:'hello'}));console.log(wrapper.find(_reactNative.Text));expect(wrapper.find(_reactNative.Text).text()).toEqual('hello');});});
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIkhlYWRlci50ZXN0LmpzIl0sIm5hbWVzIjpbImNvbmZpZ3VyZSIsImFkYXB0ZXIiLCJkZXNjcmliZSIsIml0IiwidHJlZSIsImNyZWF0ZSIsInRvSlNPTiIsImV4cGVjdCIsInRvTWF0Y2hTbmFwc2hvdCIsIndyYXBwZXIiLCJ0b0hhdmVMZW5ndGgiLCJjb25zb2xlIiwibG9nIiwiZmluZCIsInRleHQiLCJ0b0VxdWFsIl0sIm1hcHBpbmdzIjoiQUFBQSx5Q0FFQSw0QiwyQ0FDQSxxQixpQ0FFQSxzRCxtRUFDQSw4Qiw2Q0FFQSwyRCx3SkFDQSxpQkFBT0EsU0FBUCxDQUFpQixDQUFFQyxRQUFTLGtDQUFYLENBQWpCLEVBRUFDLFNBQVMsUUFBVCxDQUFtQixVQUFNLENBQ3ZCQyxHQUFHLG1CQUFILENBQXdCLFVBQU0sQ0FDNUIsR0FBTUMsTUFBTyw0QkFBU0MsTUFBVCxDQUNYLDhDQURXLEVBRVhDLE1BRlcsRUFBYixDQUdBQyxPQUFPSCxJQUFQLEVBQWFJLGVBQWIsR0FDRCxDQUxELEVBTUFMLEdBQUcsbUNBQUgsQ0FBd0MsVUFBTSxDQUM1QyxHQUFNTSxTQUFVLG9CQUFRLHFEQUFSLENBQWhCLENBQ0FGLE9BQU9FLE9BQVAsRUFBZ0JDLFlBQWhCLENBQTZCLENBQTdCLEVBQ0QsQ0FIRCxFQUlBUCxHQUFHLG1DQUFILENBQXdDLFVBQU0sQ0FDNUMsR0FBTU0sU0FBVSxvQkFBUSxxREFBUixDQUFoQixDQUNBRixPQUFPRSxPQUFQLEVBQWdCQyxZQUFoQixDQUE2QixDQUE3QixFQUNELENBSEQsRUFJQVAsR0FBRyxpREFBSCxDQUFzRCxVQUFNLENBQzFELEdBQU1NLFNBQVUsb0JBQVEsMENBQVEsTUFBTSxPQUFkLEVBQVIsQ0FBaEIsQ0FDQUUsUUFBUUMsR0FBUixDQUFZSCxRQUFRSSxJQUFSLG1CQUFaLEVBQ0FOLE9BQU9FLFFBQVFJLElBQVIsb0JBQW1CQyxJQUFuQixFQUFQLEVBQWtDQyxPQUFsQyxDQUEwQyxPQUExQyxFQUNELENBSkQsRUFLRCxDQXBCRCIsImZpbGUiOiJIZWFkZXIudGVzdC5qcyIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAncmVhY3QtbmF0aXZlJztcbmltcG9ydCB7IFRleHQsIFZpZXcgfSBmcm9tICdyZWFjdC1uYXRpdmUnO1xuaW1wb3J0IFJlYWN0IGZyb20gJ3JlYWN0JztcbmltcG9ydCBIZWFkZXIgZnJvbSAnLi4vJztcblxuaW1wb3J0IHJlbmRlcmVyIGZyb20gJ3JlYWN0LXRlc3QtcmVuZGVyZXInO1xuaW1wb3J0IHsgc2hhbGxvdyB9IGZyb20gJ2VuenltZSc7XG5pbXBvcnQgRW56eW1lIGZyb20gJ2VuenltZSc7XG5pbXBvcnQgRW56eW1lQWRhcHRlciBmcm9tICdlbnp5bWUtYWRhcHRlci1yZWFjdC0xNic7XG5Fbnp5bWUuY29uZmlndXJlKHsgYWRhcHRlcjogbmV3IEVuenltZUFkYXB0ZXIoKSB9KTtcblxuZGVzY3JpYmUoJ0hlYWRlcicsICgpID0+IHtcbiAgaXQoJ3JlbmRlcnMgY29ycmVjdGx5JywgKCkgPT4ge1xuICAgIGNvbnN0IHRyZWUgPSByZW5kZXJlci5jcmVhdGUoXG4gICAgICA8SGVhZGVyLz5cbiAgICApLnRvSlNPTigpO1xuICAgIGV4cGVjdCh0cmVlKS50b01hdGNoU25hcHNob3QoKTtcbiAgfSk7XG4gIGl0KCdpdCBzaG91bGQgcmVuZGVyIDEgdGV4dCBjb21wb25lbnQnLCAoKSA9PiB7XG4gICAgY29uc3Qgd3JhcHBlciA9IHNoYWxsb3coPFRleHQvPik7XG4gICAgZXhwZWN0KHdyYXBwZXIpLnRvSGF2ZUxlbmd0aCgxKTtcbiAgfSk7XG4gIGl0KCdpdCBzaG91bGQgcmVuZGVyIDEgdmlldyBjb21wb25lbnQnLCAoKSA9PiB7XG4gICAgY29uc3Qgd3JhcHBlciA9IHNoYWxsb3coPFZpZXcvPik7XG4gICAgZXhwZWN0KHdyYXBwZXIpLnRvSGF2ZUxlbmd0aCgxKTtcbiAgfSk7XG4gIGl0KCdpdCBzaG91bGQgcmVuZGVyIHRpdGxlIGFzIGhlbGxvIHdoZW4gZ2l2ZW4gcHJvcCcsICgpID0+IHtcbiAgICBjb25zdCB3cmFwcGVyID0gc2hhbGxvdyg8SGVhZGVyIHRpdGxlPSdoZWxsbycvPik7XG4gICAgY29uc29sZS5sb2cod3JhcHBlci5maW5kKFRleHQpKTtcbiAgICBleHBlY3Qod3JhcHBlci5maW5kKFRleHQpLnRleHQoKSkudG9FcXVhbCgnaGVsbG8nKTtcbiAgfSk7XG59KTtcbiJdfQ==