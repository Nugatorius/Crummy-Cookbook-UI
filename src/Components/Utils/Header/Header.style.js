const styles = {
  viewStyle: {
    backgroundColor: '#F8F8F8',
    justifyContent: 'center',
    alignItems: 'center',
    height: 80,
    paddingTop: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.5,
    elevation: 2,
    position: 'relative'
  },
  textStyle: {
    fontSize: 20,
    fontWeight: '400'
  }
};

export default styles;
