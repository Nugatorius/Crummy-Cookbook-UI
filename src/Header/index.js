import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import headerStyles from './Header.style';

const Header = (props) => {
  const { textStyle, viewStyle } = headerStyles;
  return (
    <View style={viewStyle}>
      <Text style={textStyle}>{props.title}</Text>
    </View>
  );
};

Header.propTypes = {
  title: PropTypes.string
};

export default Header;
