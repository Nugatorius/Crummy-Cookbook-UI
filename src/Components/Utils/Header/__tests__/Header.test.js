import 'react-native';
import { Text, View } from 'react-native';
import React from 'react';
import Header from '../';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new EnzymeAdapter() });

describe('Header', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Header/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('it should render 1 text component', () => {
    const wrapper = shallow(<Text/>);
    expect(wrapper).toHaveLength(1);
  });
  it('it should render 1 view component', () => {
    const wrapper = shallow(<View/>);
    expect(wrapper).toHaveLength(1);
  });
  it('it should render title as hello when given prop', () => {
    const wrapper = shallow(<Header title='hello'/>);
    expect(wrapper.props().children.props.children).toBe('hello');
  });
});
