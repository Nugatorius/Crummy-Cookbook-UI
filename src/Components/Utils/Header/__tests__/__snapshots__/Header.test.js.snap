// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "elevation": 2,
      "height": 80,
      "justifyContent": "center",
      "paddingTop": 15,
      "position": "relative",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "400",
      }
    }
  />
</View>
`;
